src/Features/WcPayPromotion/Init.php

```diff
@@ -55,6 +69,16 @@ public function __construct() {
 		);
 	}
 
+	/**
+	 * Get class instance.
+	 */
+	public static function get_data_source_poller_instance() {
```

Comment:
I think instead of these intermediary functions, we could create static `$instance`'s so we can directly access the data source poller.  E.g., `WcPayPromotionDataSourcePoller::get_instance()`.

Reply:
Yeah not a bad idea, in that case I might change the `DataSourcePoller` to an abstract class, and extend it to each place with the `get_instance` static function.